File: sourcecode/aspnet-core-3-api/Entities/Notification.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace WebApi.Entities
{
    public enum NotificationType
    {
        Posted,
        ReactedPost,       
        Commented,
        FollowRequest,
        FollowRequestAwser,
        ReactedComment
    }
    public enum Status
    {
        Created,
        Seen,
        Opened,
        Sent, //đã gửi
        Accepted,
        Declined,
        Public,
        Private
    }
    public class Notification
    {
        public int Id { get; set; }
        public int ActionOwnerId { get; set; }
        public NotificationType NotificationType { get; set; }
        public int? PostId { get; set; }
        public int? CommentId { get; set; }
        public int ReiceiverId { get; set; }
        public DateTime Created { get; set; }
        public Status  Status { get; set; }
    }
}

File: sourcecode/Angular/client/src/app/_models/notification.ts
export interface Notification {
  id: number;
  actionOwnerId: number;
  notificationType: number;
  postId?: number;
  CommentId?: number;
  reiceiverId: number;
  created: Date;
  status: number;
  actionOwnerName: string;
  actionOwnerAvatarPath: string;
  reiceiverName: string;
}
